src/status_providers/github.ts

```diff
@@ -79,10 +71,34 @@ export class GithubStatusProvider extends BaseStatusProvider {
       logger.info(
         'No config provided for Github status provider, calculating the combined status...'
       );
-      const results = [
-        this.getResultFromRevisionStatus(revisionStatus),
-        this.getResultFromRevisionChecks(revisionChecks),
-      ];
+
+      let commitApiStatusResult;
+      if (
+        revisionStatus.total_count === 0 &&
+        revisionStatus.state === 'pending'
+      ) {
+        // Edge case, this is what GitHub returns when there are no registered legacy checks.
+        logger.debug('No legacy checks detected, checking for runs...');
+        if (revisionChecks.total_count > 0) {
+          logger.debug('Check runs exist, continuing...');
+          commitApiStatusResult = CommitStatus.SUCCESS;
+        } else {
+          logger.warn('No valid build contexts detected, did any checks run?');
+          return CommitStatus.FAILURE;
```

Comment:
Wouldn't it be better to do
```js
commitApiStatusResult = CommitStatus.NOT_FOUND
```

and let the rest of the flow take care of that (around line 106 below)?

Reply:
Yes, makes sense 👍